### DISCUSSION TITLE
Scenario "Unimak Island": Fun with Mr Jason

### DISCUSSION THREAD

**Discussion by fduran** [SadServers/sadservers#33] [Scenarios]
Discussions about the scenario [Unimak Island](https://github.com/fduran/sadservers/tree/main/scenarios/unimak)

**Comment by ghost** [comment]
I still get that I did not solve this challenge but I think the solution in the check.sh is wrong 
See my output 

jq '.data.stations[] | select(.station_id == "05c5e17c-7aa9-49b7-9da3-9db4858ec1fc")' station_information.json
{
  "eightd_has_key_dispenser": false,
  "rental_methods": [
    "KEY",
    "CREDITCARD"
  ],
  "external_id": "05c5e17c-7aa9-49b7-9da3-9db4858ec1fc",
  "station_id": "05c5e17c-7aa9-49b7-9da3-9db4858ec1fc",
  "rental_uris": {
    "ios": "https://bkn.lft.to/lastmile_qr_scan",
    "android": "https://bkn.lft.to/lastmile_qr_scan"
  },
  "region_id": "71",
  "capacity": 32,
  "short_name": "6569.09",
  "electric_bike_surcharge_waiver": false,
  "has_kiosk": false,
  "name": "W 35 St & 9 Ave",
  "lon": -73.9960889518261,
  "lat": 40.75414519263519,
  "station_type": "classic",
  "eightd_station_services": []
}
admin@i-0cdd7769078d36c89:~$ jq '.data.stations[] | select(.station_id == "8d8414808b15d55dad857fd5aeb2aebc")' station_information.json
admin@i-0cdd7769078d36c89:~$ cd agent
admin@i-0cdd7769078d36c89:~/agent$ cat check.sh 
#!/usr/bin/bash
res=$(md5sum /home/admin/mysolution |awk '{print $1}')
res=$(echo $res|tr -d '\r')

if [[ "$res" = "8d8414808b15d55dad857fd5aeb2aebc" ]]
then
  echo -n "OK"
else
  echo -n "NO"
fi

**Reply by fduran** [reply]
Thanks for your feedback. The solution station_id is `05c5e17c-7aa9-49b7-9da3-9db4858ec1fc` as you found out, and its md5 hash is the one that check.sh validates:

```
echo "05c5e17c-7aa9-49b7-9da3-9db4858ec1fc" |md5
8d8414808b15d55dad857fd5aeb2aebc
```

**Reply by ghost** [reply]
hmm, then I have to look why I still get a message that my solution is not right.

**Comment by ghost** [comment]
I see another  output : 

```
cat mysolution 
"05c5e17c-7aa9-49b7-9da3-9db4858ec1fc"
admin@i-09d19a385d7a279f6:~$ md5sum ~/mysolution
fd95ff610ad13bb45fba5717df25a7a4  /home/admin/mysolution
```

**Reply by fduran** [reply]
Hello, you are wrapping the solution in double quotes, ie `cat mysolution` should return `05c5e17c-7aa9-49b7-9da3-9db4858ec1fc` not `"05c5e17c-7aa9-49b7-9da3-9db4858ec1fc"`. You can test this: `echo '"05c5e17c-7aa9-49b7-9da3-9db4858ec1fc"' |md5
fd95ff610ad13bb45fba5717df25a7a4`. Maybe I'll clarify that the solution should be the UUID without quotes.

**Comment by ghost** [comment]
Can you take a look what I do wrong here ?
